src/WorkflowUpdate/WorkflowUpdate.workflow.cs

```diff
@@ -0,0 +1,71 @@
+namespace TemporalioSamples.WorkflowUpdate;
+
+using Temporalio.Workflows;
+
+[Workflow]
+public class WorkflowUpdate
+{
+    private readonly Queue<UiRequest> requests = new();
+    private ScreenId screen = ScreenId.Screen1;
+    private bool updateInProgress;
+
+    [WorkflowRun]
+    public async Task RunAsync()
```

Comment:
Technically it may be clearer to have the update handler have its own logic and if it needs to have a homemade lock to prevent other updates from running using a bool field like you have, so be it, and the run function just be `public Task RunAsync() => Workflow.WaitConditionAsync(() => CurrentScreen == ScreenId.End)`, but what you have here is fine too.